### PR TITLE
Device repo selection of the New Device Onboarding

### PR DESCRIPTION
**Pull request by ryaplots**
#### Summary

Refers to #4847 
New device onboarding - Device Type Repository Section (first selection).

#### Changes

- Removes reducer and refs;
- Using context;

#### Testing

Manually

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/webui/console/containers/device-onboarding-form/device-type-form-section/device-type-repository-form-section/device-card/connect.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// Copyright © 2021 The Things Network Foundation, The Things Industries B.V.
```
Comment (kschiffer) [inline pkg/webui/console/containers/device-onboarding-form/device-type-form-section/device-type-repository-form-section/device-card/connect.js]:
```suggestion
// Copyright © 2022 The Things Network Foundation, The Things Industries B.V.
```

Comment (kschiffer) [inline pkg/webui/console/containers/device-onboarding-form/device-type-form-section/device-type-repository-form-section/device-card/connect.js]:
```suggestion
// Copyright © 2022 The Things Network Foundation, The Things Industries B.V.
```
Please update on all files.

File: pkg/webui/console/containers/device-onboarding-form/device-type-form-section/device-type-repository-form-section/device-card/device-card.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+  return label
+}
+
+const getLorawanPhyVersionLabel = ({ lorawan_phy_version }) => {
+  const { label } =
+    LORAWAN_PHY_VERSIONS.find(version => version.value === lorawan_phy_version) || {}
+
+  return label
+}
```
Comment (kschiffer) [inline pkg/webui/console/containers/device-onboarding-form/device-type-form-section/device-type-repository-form-section/device-card/device-card.js]: Let's move this to the device utils. We can use it on the end device overview as well.

File: pkg/webui/console/containers/device-onboarding-form/index.js [hunk 3]
```diff
@@ -52,7 +59,21 @@ const DeviceOnboardingForm = props => {
...
 
-export default DeviceOnboardingForm
+export default connect(
+  state => ({
+    appId: selectSelectedApplicationId(state),
+  }),
+  dispatch => ({
+    getRegistrationTemplate: (appId, version) => dispatch(getTemplate(appId, version)),
+  }),
+)(DeviceOnboardingForm)
```
Comment (kschiffer) [inline pkg/webui/console/containers/device-onboarding-form/index.js]: For consistency, consider adding that to a seperate `connect.js` module.

**Review by kschiffer — CHANGES_REQUESTED** [review]
LGTM apart from some minor issues.
One thing - can you try adding something that didn't make it into the last version:
When making a selection, the focus should automatically jump to the next available select input (if possible).

**Comment by ryaplots** [comment]
> LGTM apart from some minor issues. One thing - can you try adding something that didn't make it into the last version: When making a selection, the focus should automatically jump to the next available select input (if possible).

@kschiffer You mean like this:

https://user-images.githubusercontent.com/56658938/178702905-a2f74fe4-5a8c-4870-a772-4d7597ab885b.mov

**Comment by kschiffer** [comment]
Perfect

**Review by kschiffer — APPROVED** [review]

**Pull request merged**